lib/integrations/awsoidc/create_ec2ice.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/gravitational/trace"
+)
+
+// CreateEC2ICERequest contains the required fields to create an AWS EC2 Instance Connect Endpoint.
+type CreateEC2ICERequest struct {
+	// Cluster is the Teleport Cluster Name.
+	// Used to tag resources created in AWS.
+	Cluster string
+
+	// IntegrationName is the integration name.
+	// Used to tag resources created in AWS.
+	IntegrationName string
+
+	// SubnetID is the Subnet where the Endpoint will be created.
+	SubnetID string
+
+	// SecurityGroupIDs is a list of SecurityGroups to assign to the Endpoint.
+	// If an empty list is provided, the Endpoint will receive the default SG for the Subnet's VPC.
+	SecurityGroupIDs []string
+
+	// ResourceCreationTags is used to add tags when creating resources in AWS.
+	// Defaults to:
+	// - teleport.dev/cluster: <cluster>
+	// - teleport.dev/origin: aws-oidc-integration
+	// - teleport.dev/integration: <integrationName>
+	ResourceCreationTags awsTags
+}
+
+// CheckAndSetDefaults checks if the required fields are present.
+func (req *CreateEC2ICERequest) CheckAndSetDefaults() error {
+	if req.Cluster == "" {
+		return trace.BadParameter("cluster is required")
+	}
+
+	if req.IntegrationName == "" {
+		return trace.BadParameter("integration is required")
+	}
+
+	if req.SubnetID == "" {
+		return trace.BadParameter("subnet id is required")
+	}
+
+	if len(req.ResourceCreationTags) == 0 {
+		req.ResourceCreationTags = DefaultResourceCreationTags(req.Cluster, req.IntegrationName)
```

Comment:
`DefaultResourceCreationTags` doesn't appear to be used outside package `awsoidc` - can we unexport it?